src/jarabe/screenshotpanel/gui.py

```diff
@@ -0,0 +1,341 @@
+# Copyright (C) 2008 One Laptop Per Child
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+import os
+import dbus
+import cairo
+import logging
+import StringIO
+import tempfile
+from gettext import gettext as _
+
+from gi.repository import GObject
+from gi.repository import Gtk
+from gi.repository import Gdk
+from gi.repository import Gio
+
+from sugar3 import env
+
+from sugar3.datastore import datastore
+from sugar3.graphics.icon import Icon
+from sugar3.graphics import style
+from sugar3.graphics.alert import Alert, TimeoutAlert
+from sugar3.graphics import iconentry
+
+
+from jarabe.model.session import get_session_manager
+from jarabe.screenshotpanel.toolbar import MainToolbar
+from jarabe import config
+from jarabe.model import shell
+
+
+_logger = logging.getLogger('ScreenshotPanel')
+
+
+class ScreenshotPanel(Gtk.Window):
+    '''
+    Generates a pop papel to allow user to save the
+    screenshot by the name of his choice
+    '''
+
+    __gtype_name__ = 'ScreenshotPanel'
+
+
+    def __init__(self):
+        Gtk.Window.__init__(self)
```

Comment:
Most of ```__init__``` seems remarkably familiar; is there an opportunity to refactor other similar classes?